pkg/kube/controllers/extendedjob/job_reconciler.go

```diff
@@ -154,7 +154,25 @@ func (r *ReconcileJob) jobPod(ctx context.Context, name string, namespace string
 	if len(list.Items) == 0 {
 		return nil, errors.Errorf("Job %s does not own any pods?", name)
 	}
-	return &list.Items[0], nil
+
+	// If there is only one jobpod, then return index 0 pod
+	if len(list.Items) == 1 {
+		return &list.Items[0], nil
+	} else {
+		// If there are more than one jobpods, then return the earliest
+		// created jobpod. There will be multiple jobpods when jobpods
```

Comment:
Why would you want the earliest created pod returned? I'd have thought that you'd be more interested in the latest, possibly succeeded one.

Do you mind adding a test for this change?

Reply:
Sure, I will add a test for it. 

Yes, it makes sense to check for the latest change. But by reading the code of the `job_controller`, I realized we only care about the first job pod. In this line [0], the `o.Status.Failed` count is only 1, but when there is a failed test, the count will be 2. When the count is 2, we don't reconcile as per the condition.

Let me know if we should change this ?

[0] https://github.com/cloudfoundry-incubator/cf-operator/blob/master/pkg/kube/controllers/extendedjob/job_controller.go#L57